x-pack/plugins/security_solution/public/common/hooks/use_experimental_features.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useSelector } from 'react-redux';
+import { State } from '../../common/store';
+import { ExperimentalFeatures } from '../../../common/experimental_features';
+
+export const useIsExperimentalFeatureEnabled = (feature: keyof ExperimentalFeatures): boolean => {
+  return useSelector(({ app: { enableExperimental } }: State) =>
+    enableExperimental ? enableExperimental[feature] : false
```

Comment:
I'm wondering if instead of defaulting to `false` we should just `throw` here if the `feature` is not part of `EnableExperimental`? something like:


```typescript
if (!(feature in enableExperimental)) {
    throw new Error(`invalid enable value ${feature}`);
}
return enableExperimental[feature];
```

_(you can use the checker utility that is part of the `enableExperimental` code that I pushed through in prior commit to validate the feature key)_


A few reasons come to mind:

1. We should not assume that the default value is `false` - perhaps tehre will be cases in the future with the default is `true`
2. we should not have code trying to use experimental feature flags when those are removed, so `throw`'ing here would surface that
3. this hook could potentially return `undefined` if `feature` argument is not actually part of `EnableExperimental` (ex. via casting as you did in the tests or using ts-ignore)

Reply:
Good catch!
1. Totally agree.
2. Agreed.
3. It's an edge case but yes, it could happen.

I'll do the changes to throw an error when the feature flag doesn't exists 👍